plugins/node/opentelemetry-instrumentation-knex/src/instrumentation.ts

```diff
@@ -45,8 +59,21 @@ const DEFAULT_CONFIG: KnexInstrumentationConfig = {
 };
 
 export class KnexInstrumentation extends InstrumentationBase<KnexInstrumentationConfig> {
+  private _semconvStability: SemconvStability = SemconvStability.OLD;
+
   constructor(config: KnexInstrumentationConfig = {}) {
     super(PACKAGE_NAME, PACKAGE_VERSION, { ...DEFAULT_CONFIG, ...config });
+
+    for (const entry of getStringListFromEnv('OTEL_SEMCONV_STABILITY_OPT_IN') ??
+      []) {
+      if (entry.toLowerCase() === 'database/dup') {
+        // database/dup takes highest precedence. If it is found, there is no need to read the rest of the list
+        this._semconvStability = SemconvStability.DUPLICATE;
+        break;
+      } else if (entry.toLowerCase() === 'database') {
+        this._semconvStability = SemconvStability.STABLE;
+      }
```

Comment:
FYI: Not for now (unless this PR waits for a bit), but there is a (not yet released) utility method being added to `@opentelemetry/instrumentation` for this: `databaseSemconvStabilityFromStr(process.env.OTEL_SEMCONV_STABILITY_OPT_IN)`. See https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/opentelemetry-instrumentation/src/semconvStability.ts#L17

This will be in `@opentelemetry/instrumentation@2.1.0`. The intent is to release this soon (perhaps this week).

Reply:
I can wait for that, although I'll push the other changes on my branch.

Reply:
I got the version wrong, it is in `@opentelemetry/instrumentation@0.201.0`, now published.